examples/next-openai-pages/pages/chat/stream-chat-files/index.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @next/next/no-img-element */
+import { useChat } from 'ai/react';
+import { useRef, useState } from 'react';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/stream-chat',
+    });
+
+  const [files, setFiles] = useState<FileList | undefined>(undefined);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.experimental_files?.map((file, index) =>
+                  file.url.includes('image/') ? (
+                    <img
+                      key={`${message.id}-${index}`}
+                      className="w-24 rounded-md"
+                      src={file.url}
+                      alt="image"
+                    />
+                  ) : file.url.includes('text/') ? (
```

Comment:
can we rely on the type here instead? in urls it could also be e.g. the path

Reply:
done!